realm/src/androidTest/java/io/realm/RealmConfigurationTest.java

```diff
@@ -416,4 +417,20 @@ public void testDifferentDirSameName() {
         realm1.close();
         realm2.close();
     }
+
+    public void testKeyStorage() throws Exception {
+        // Generate a key and use it in a RealmConfiguration
+        byte[] oldKey = TestHelper.getRandomKey(12345);
+        byte[] key = oldKey;
+        RealmConfiguration config = new RealmConfiguration.Builder(getContext()).encryptionKey(key).build();
+
+        // Generate a different key and assign it to the same variable
+        byte[] newKey = TestHelper.getRandomKey(67890);
+        MoreAsserts.assertNotEqual(key, newKey);
+        key = newKey;
```

Comment:
isn't the problem more if you do `key[0] = 42`, i.e changing the value inside the array?